receiver/prometheusreceiver/metrics_receiver.go

```diff
@@ -288,6 +292,15 @@ func (r *pReceiver) initPrometheusComponents(ctx context.Context, logger log.Log
 	return nil
 }
 
+var _ prometheus.Registerer = &noopRegistry{}
+
+// noopRegistry implements the prometheus.Registerer interface by not doing anything.
+type noopRegistry struct{}
```

Comment:
I'm not sure this is the correct approach.  The `prometheus.DefaultRegisterer` is already in use today and I'm not sure I see a good reason to prevent this component from utilizing it simply because one of the applications it is used it does not use it.  I agree that it would be good to give the user control over where these metrics are produced, but this action seems to simply eliminate any choice they may have at this time and prevent anyone who may have wanted to use these metrics from doing so.

Reply:
I think we need to get that fixed first, then.  Since we know what the problem and fix are upstream let's fix it there and not introduce an unnecessary regression here.

Reply:
Ok, reverted to a draft.

Reply:
I've created https://github.com/prometheus/prometheus/pull/13610 to fix this.  Hopefully this approach is acceptable.

Reply:
I've added a receiver name, but it doesn't help with tests since the receiver name is always "noop".  It probably needs more work before this will pass tests

Reply:
Fixed.